lib/event_sourcery/event_processing/esp_runner.rb

```diff
@@ -65,8 +67,16 @@ def shutdown
         @shutdown_requested = true
       end
 
-      def wait_till_shutdown_requested
-        sleep(1) until @shutdown_requested
+      def wait_until_shutdown
+        loop do
+          sleep(1)
+          break if @shutdown_requested
+          break if a_process_has_terminated? && @shutdown_if_child_process_fails
```

Comment:
Perhaps we should log something in this scenario so it's obvious what's happening